From 1a3897db8a1c60729ce7620502c65522404dbb53 Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Mon, 16 Sep 2024 19:01:56 +0300
Subject: [PATCH] [FSW-8694] fs1: check invalid flash offset

With this patch, added support to check desc spi
offset is valid if not report with proper
progress bit.

Change-Id: Iccc16bd0a3d814e86061ed0b9c724e40f5ecfbaf
---

diff --git a/common/boot_flow.c b/common/boot_flow.c
index c272687..a435328 100644
--- a/common/boot_flow.c
+++ b/common/boot_flow.c
@@ -43,6 +43,11 @@
 		return BOOT_ROM_FAIL;
 	}
 
+	if (desc->spi_offset[manifest_idx] >= HL_FLASH_SIZE) {
+		boot_rom_status_update_bringup(img_sts_base, PROGRESS_BRINGUP_INVALID_FLASH_OFFSET);
+		return BOOT_ROM_FAIL;
+	}
+
 	/* Validate Image descriptor start address and other ranges */
 	if (boot_rom_verify_img_desc_address_range(desc, img_sts_base, img_type)
 	    != BOOT_ROM_SUCCESS)
diff --git a/common/status.h b/common/status.h
index 3f06c23..842a135 100644
--- a/common/status.h
+++ b/common/status.h
@@ -22,6 +22,7 @@
 	PROGRESS_BRINGUP_UNALIGNED_DEST_ADDR,
 	PROGRESS_BRINGUP_INVALID_IMG_SIZE,
 	PROGRESS_BRINGUP_UNALIGNED_IMG_SIZE,
+	PROGRESS_BRINGUP_INVALID_FLASH_OFFSET,
 	PROGRESS_BRINGUP_DMA_ERROR,
 	PROGRESS_BRINGUP_DMA,
 	PROGRESS_BRINGUP_KEY_AUTH_FAIL,
